### PR TITLE
test: add integration tests for restoreFromPlayer process death

### PR DESCRIPTION
**Pull request by AgiMaulana**
## Summary
- Add `PlaybackManagerProcessDeathIntegrationTest` with 5 test cases covering SEARCH, DEFAULT, PINNED context restoration, nextPage estimation, and empty player edge case
- Expose internal getters in `PlaybackManager` for `loadedIds`, `playbackContext`, and `nextPage` for testing
- Update existing unit tests to use renamed private fields (`_loadedIds`, `_playbackContext`)
- Mark `integration-process-death-tests` as done in `playback-queue-plan.md`

## Tests
- 8 unit tests pass (5 new + 2 existing + 1 RadioMediaItemTest)

### CODE REVIEW

File: core/radioplayer/src/test/kotlin/io/github/agimaulana/radio/core/radioplayer/internal/PlaybackManagerTest.kt [hunk 1]
```diff
@@ -93,7 +93,7 @@ class PlaybackManagerTest {
         assertEquals(3, nextPageValue)
 
         // playbackContext should be SEARCH with query "gen"
-        val playbackContextField = PlaybackManager::class.java.getDeclaredField("playbackContext")
+        val playbackContextField = PlaybackManager::class.java.getDeclaredField("_playbackContext")
```
[the remark this answers is not in the document]
Comment (AgiMaulana) [reply core/radioplayer/src/test/kotlin/io/github/agimaulana/radio/core/radioplayer/internal/PlaybackManagerTest.kt:96]: make the `playbackContext` as private as possible

File: core/radioplayer/src/test/kotlin/io/github/agimaulana/radio/core/radioplayer/internal/PlaybackManagerTest.kt [hunk 2]
```diff
@@ -48,7 +48,7 @@ class PlaybackManagerTest {
         val manager = PlaybackManager(player) { nextPage, query -> fetched }
 
         // simulate that id1 was already loaded to test dedupe
-        val loadedIdsField = PlaybackManager::class.java.getDeclaredField("loadedIds")
+        val loadedIdsField = PlaybackManager::class.java.getDeclaredField("_loadedIds")
```
[the remark this answers is not in the document]
Comment (AgiMaulana) [reply core/radioplayer/src/test/kotlin/io/github/agimaulana/radio/core/radioplayer/internal/PlaybackManagerTest.kt:51]: make the `loadedIds ` as private as possible

[the remark this answers is not in the document]
Comment (AgiMaulana) [reply core/radioplayer/src/test/kotlin/io/github/agimaulana/radio/core/radioplayer/internal/PlaybackManagerTest.kt:51]: `_loadedIds` itself is private

File: core/radioplayer/src/test/kotlin/io/github/agimaulana/radio/core/radioplayer/internal/PlaybackManagerTest.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (AgiMaulana) [reply core/radioplayer/src/test/kotlin/io/github/agimaulana/radio/core/radioplayer/internal/PlaybackManagerTest.kt:96]: `_playbackContext ` itself is private
